lib/vsc/mympirun/mpi/openmpi.py

```diff
@@ -219,3 +230,24 @@ class OpenMpi3(OpenMpiOversubscribe):
         'shm': 'vader,self',
         'socket': 'vader,tcp,self',
     }
+
+
+class OpenMpi4(OpenMpiOversubscribe):
+
+    """
+    An implementation of the MPI class for OpenMPI 4.x & more recent.
+    """
+
+    _mpirun_version = staticmethod(lambda ver: version_in_range(ver, '4', None))
+
+    def use_ucx_pml(self):
+        """Determine whether or not to use the UCX Point-to-Point Messaging Layer (PML)."""
```

Comment:
why is this liited to openmpi 4? any openmpi with ucx support should probably use

Reply:
Several reasons:

* OpenMPI 4 (used in `2020a` toolchains) is a natural cutoff, since it's a new major version (`2019b` and older use OpenMPI <=3)
* "OpenMPI supports UCX starting from version 3.0, but it’s recommended to use version 4.0 or higher due to stability and performance improvements." (see https://openucx.github.io/ucx/running.html)
* The OpenMPI 3.1.4 modules we have in `foss/2019b` have UCX support, but rely on the `ucx` RPM we have installed in the OS (while in `foss/2020a` we install `UCX` as a proper dep for OpenMPI 4.0.3, so we're in control)
* We've been using `openib` rather than `ucx` for a while now in `foss/2019b`, which mostly works fine, so why change that...
* `mympirun` is currently broken with `foss/2020a`, because the OpenMPI in there no longer supports `openib` (see https://github.com/easybuilders/easybuild-easyblocks/pull/2188), this PR fixes that by using `ucx` instead

We can still consider also using the `ucx` PML with OpenMPI 3.x, but then we need to reinstall the OpenMPI 3 modules with a proper UCX dep (so we're in control), and do more thorough testing.

Reply:
i thought we couldn't build the `openib` due to missing headers, and that it was thus also not avail on openmpi3.
can you also add the 2nd item in a comment or in docstring?

Reply:
OpenMPI 3 was being configured with `--without-verbs` on RHEL8 because of a bug in EasyBuild which is now fixed (see https://github.com/easybuilders/easybuild-framework/pull/3477), but we still need to reinstall OpenMPI 3 on `doduo`, that's on our TODO list before we give pilot users access.